packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -0,0 +1,234 @@
+import useListBox from "../../../useListBox";
+import applyOnChange from "../../../helpers/applyOnChange";
+
+function getOnChangeFn(state) {
+  return applyOnChange(state.onChange, "listbox:option-selected");
+}
+
+function selectSingleOption({ activeOptionIndex, isOpen, state, dispatch }) {
+  dispatch({
+    type: useListBox.types.selectSingleOption,
+    payload: { activeOptionIndex, isOpen, onChangeFn: getOnChangeFn(state, activeOptionIndex) },
+  });
+}
+
+function selectMultipleOption({ activeOptionIndex, state, dispatch }) {
+  dispatch({
+    type: useListBox.types.selectMultipleOption,
+    payload: { activeOptionIndex, onChangeFn: getOnChangeFn(state, activeOptionIndex) },
+  });
+}
+
+export function isOptionSelected(state, index) {
+  return state.selectedOptions.includes(index);
+}
+
+export function isOptionVisible(state, key) {
+  if (!Object.keys(state.options).length) return;
+
+  const keyInt = Number.parseInt(key, 10);
+
+  if (typeof state.options[keyInt] === "undefined") {
+    return true;
+  }
+
+  if (state.noResultsFound) {
+    return false;
+  }
+
+  if (state.options[keyInt].isHidden) {
+    return false;
+  }
+
+  return !state.filteredOptions.length || state.filteredOptions.includes(keyInt);
+}
+
+function isDisabled(state, key) {
+  return state.options[key].isDisabled;
+}
+
+export function getNextOptionActiveIndex(state, isAscending = true) {
+  const { activeOption, filteredOptions, options } = state;
+  const optionsKeys = Object.keys(state.options);
+
+  if (state.noResultsFound) return null;
+
+  if (state.activeOption === null) return 0;
+
+  if (optionsKeys.length === 1) {
+    return 0;
+  }
+
+  if (optionsKeys.length - 1 < activeOption) {
+    return 0;
+  }
+
+  let key = state.options[activeOption].index;
+
+  let index = null;
+  let keepIterating = true;
+
+  while (keepIterating) {
+    if (isAscending) {
+      if (filteredOptions.length === 1) {
+        keepIterating = false;
+        return filteredOptions[0];
+      }
+
+      if ((filteredOptions.length && key + 1 > options.length) || key + 1 > optionsKeys.length - 1) {
+        keepIterating = false;
+        return null;
+      }
+
+      key++;
+    } else {
+      if ((filteredOptions.length && key - 1 < 0) || key - 1 < 0) {
+        keepIterating = false;
+        return null;
+      }
+
+      key--;
+    }
+
+    if (isOptionVisible(state, key) && !isDisabled(state, key)) {
+      index = key;
```

Comment:
Can you just `return key` here, then eliminate the return on line 99 (as well as the `keepIterating` variable)?